test/servers/binary_framed_test.exs

```diff
@@ -0,0 +1,166 @@
+defmodule Servers.BinaryFramedIntegrationTest do
+  use ThriftTestCase
+
+  @thrift_file name: "server_test.thrift", contents: """
+    exception TestException {
+      1: string message,
+      2: i32 code,
+    }
+
+    exception UserNotFound {
+      1: string message,
+    }
+
+    exception OtherException {
+      2: string message,
+    }
+
+    struct IdAndName {
+     1: i64 id,
+     2: string name,
+    }
+
+    service ServerTest {
+      void returns_nothing()
+      oneway void do_async(1: string message);
+      bool ping();
+      bool checked_exception() throws (1: TestException ex);
+      bool multiple_exceptions(1: i32 exc_type) throws
+         (1: TestException e, 2: UserNotFound unf, 3: OtherException other);
+      bool server_exception();
+      IdAndName echo_struct(1: IdAndName id_and_name);
+      i64 myCamelCasedFunction(1: string myUserName);
+    }
+  """
+
+  def define_handler do
+    defmodule ServerTestHandler do
+      alias Servers.BinaryFramedIntegrationTest.ServerTest.Handler
+      alias Servers.BinaryFramedIntegrationTest.{TestException, UserNotFound, OtherException}
+      alias Servers.BinaryFramedIntegrationTest, as: T
+      @behaviour Handler.Behaviour
+
+      def do_async(message) do
+        Agent.update(:server_args, fn(_) -> message end)
+      end
+
+      def ping, do: true
+
+      def checked_exception do
+        raise %T.TestException{message: "Oh noes!", code: 400}
+      end
+
+      def server_exception do
+        raise "This wasn't supposed to happen"
+      end
+
+      def echo_struct(id_and_name), do: id_and_name
+
+      def returns_nothing, do: nil
+
+      def multiple_exceptions(exception_type) do
+        case exception_type do
+          1 -> raise %TestException{message: "BOOM", code: 124}
+          2 -> raise %UserNotFound{message: "Not here!"}
+          3 -> raise %OtherException{message: "This is the other"}
+          _ -> true
+        end
+      end
+
+      def my_camel_cased_function(user_name) do
+        Agent.update(:server_args, fn(_) -> user_name end)
+        2421
+      end
+    end
+  end
+
+  alias Servers.BinaryFramedIntegrationTest.ServerTest.Client
+  alias Servers.BinaryFramedIntegrationTest.ServerTest.Server
+  alias Thrift.TApplicationException, as: TAE
+
+  setup do
+    server_port = :rand.uniform(10000) + 12000
+
+    {:ok, agent} = Agent.start_link(fn -> nil end, name: :server_args)
+
+    on_exit fn ->
+      if Process.alive?(agent) do
+        ref = Process.monitor(agent)
+        Agent.stop(agent)
+
+          receive do
+            {:DOWN, ^ref, _, _, _} ->
+              :ok
+          end
+      end
+    end
+
+    :rand.seed(:exs64, :erlang.now)
+
+
+    {:module, mod_name, _, _} = define_handler()
+    {:ok, client} = Client.Framed.start_link("localhost", server_port, [])
+
+    {:ok, _} = Server.Framed.start_link(mod_name, server_port, [worker_count: 20])
```

Comment:
I'm not psyched about passing the handler module into the server's start_link. What's the use case for that, outside of this test? If we generate a ServerTest.Server, I think it would be reasonable to just expect the user to have defined a ServerTest.Handler and generate calls to ServerTest.Handler.ping or whatever.